File: extension/src/types/messages.ts
// CANONICAL. Mirrored in: tauri-app/src/types/messages.ts and src-tauri/src/messages.rs

/** Messages sent from Browser Extension to Desktop App */
export type ExtensionMessage =
  | { type: 'PLAY_URL'; url: string; title?: string; source: 'youtube' | 'html5' | 'local' }
  | { type: 'SYNC_STATE'; state: 'play' | 'pause' }
  | { type: 'SOURCE_LOST' };

/** Messages sent from Desktop App to Browser Extension */
export type AppMessage =
  | { type: 'READY' }
  | { type: 'STATUS'; status: 'playing' | 'paused' | 'static' | 'off' };

File: tauri-app/src-tauri/src/messages.rs
// MIRROR OF: extension/src/types/messages.ts
use serde::{Deserialize, Serialize};

#[allow(non_camel_case_types)]
#[derive(Debug, Clone, Deserialize, Serialize)]
#[serde(tag = "type")]
pub enum ExtensionMessage {
    PLAY_URL {
        url: String,
        #[serde(default)]
        title: Option<String>,
        source: String,
    },
    SYNC_STATE {
        state: String,
    },
    SOURCE_LOST,
}

#[derive(Debug, Clone, Deserialize, Serialize)]
#[serde(tag = "type")]
pub enum AppMessage {
    READY,
    STATUS {
        status: String,
    },
}
